crates/oxc_linter/src/utils/unicorn/numeric.rs

```diff
@@ -0,0 +1,13 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+
+// Determine whether this node is a decimal integer literal.
+// Copied from https://github.com/eslint/eslint/blob/cc4871369645c3409dc56ded7a555af8a9f63d51/lib/rules/utils/ast-utils.js#L1237
+lazy_static! {
+    static ref DECIMAL_INTEGER_PATTERN: Regex =
+        Regex::new(r"^(?:0|0[0-7]*[89]\d*|[1-9](?:_?\d)*)$").unwrap();
```

Comment:
Does `NumericLiteral#base` not suit your needs?

Reply:
This is to check whether 1.0100 becomes 1.01 or 1.00 becomes 1. I found that format_raw's return is_dangling_dot can distinguish this situation, so I will delete this code